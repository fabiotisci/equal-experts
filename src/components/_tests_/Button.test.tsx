import * as React from 'react';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { Button } from '../';

configure({adapter: new Adapter()});

describe('<Button>', () => {
    let clickMock: () => void;
    let wrapper: ShallowWrapper;
    
    beforeEach( () => {
        clickMock = jest.fn();        
        wrapper = shallow(<Button clicked={clickMock} classes={['test']} tabIndex={33}>My test button</Button>);
    });

    it('Should render the expected template', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.prop('children')).toEqual('My test button');
        expect(wrapper.prop('className')).toEqual('Button test');
        expect(wrapper.prop('tabIndex')).toEqual(33);
    })

    it('Should render the expected template', () => {
        wrapper.simulate('click');
        expect(clickMock).toHaveBeenCalledTimes(1);
    })        
})