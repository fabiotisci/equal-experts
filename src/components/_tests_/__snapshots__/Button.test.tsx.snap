// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> Should render the expected template 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
      Array [
        "test",
      ]
    }
    clicked={[MockFunction]}
    tabIndex={33}
  >
    My test button
  </Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "My test button",
      "className": "Button test",
      "onClick": [MockFunction],
      "tabIndex": 33,
    },
    "ref": null,
    "rendered": "My test button",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "My test button",
        "className": "Button test",
        "onClick": [MockFunction],
        "tabIndex": 33,
      },
      "ref": null,
      "rendered": "My test button",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
